mixer/pkg/runtime/config/snapshot.go

```diff
@@ -97,6 +144,49 @@ type (
 		// Instances that should be generated as part of invoking action.
 		Instances []*Instance
 	}
+
+	// Template contains info about a template
+	Template struct {
+		// Name of the template.
+		//
+		// Note this is the template's resource name and not the template's internal name that adapter developer
+		// uses to implement adapter service.
+		Name string
+
+		// InternalPackageDerivedName is the name of the template from adapter developer point of view.
+		// The service and functions implemented by the adapter is based on this name
+		// NOTE: This name derived from template proto package and not the resource name.
+		InternalPackageDerivedName string
+
+		// Template's file descriptor set.
+		FileDescSet *descriptor.FileDescriptorSet
+
+		// package name of the `Template` message
+		PackageName string
+	}
+
+	// Adapter contains info about an adapter
+	Adapter struct {
+		Name string
+
+		// Adapter's file descriptor set.
+		ConfigDescSet *descriptor.FileDescriptorSet
+
+		// package name of the `Params` message
+		PackageName string
+
+		SupportedTemplates []Template
+
+		SessionBased bool
+
+		Description string
+	}
+
+	// Client contains the connection data for the adapter
+	Client struct {
```

Comment:
I don't think this is the right place have a live connection. The config stuff should be about configuration only, not runtime state. We probably need something similar to handler table forthis.

Reply:
You are right.